src/main/java/com/google/api/codegen/go/GoContextCommon.java

```diff
@@ -34,4 +35,51 @@
     }
     return result;
   }
+
+  /**
+   * Converts and wraps the specified text into a comment block in the generated Go file.
+   *
+   * Lines are wrapped to 70 characters. See documentation of wrapLine for more details.
+   */
+  public Iterable<String> getCommentLinesWrap(String text) {
+    List<String> result = new ArrayList<>();
+    for (String line : Splitter.on(String.format("%n")).split(text)) {
+      for (String wrapped : wrapLine(line, 70)) {
+        result.add(line.isEmpty() ? "//" : "// " + wrapped);
+      }
+    }
+    return result;
+  }
+
+  /**
+   * wrapLine splits `line` into a list of lines around `length` long at whitespaces.
+   *
+   * Current implementation first splits the line into words, then add the words,
+   * separated by a single space, to the current line until the line is longer than `length`,
+   * then breaks the line.
+   * This eliminates the corner-case of `line` containing a word that is by itself longer than
```

Comment:
"corner case" (no hyphen)

I don't think a very long word in a line by itself is the problem, nor does your implementation guard against that, IIUC (if you were starting a new line anyway and the next word is super-long, then it will wind up as the only word on the line). What it does guard against is a very *short* line followed by a line longer than `length` (or more accurately, a line of length j followed by a line of length k where j+k>=`length`, which is unattractive when j << k)